### PR TITLE
Seq: Implement with support for -sw

### PR DESCRIPTION
**Pull request by yehohanan7**
Initial implementation, no support for "-f" yet.

### CODE REVIEW

File: seq/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+[package]
+name = "seq"
+version = "0.1.0"
+authors = ["John Pradeep Vincent <yehohanan7@gmail.com>"]
+license = "MPL-2.0-no-copyleft-exception"
+build = "build.rs"
+edition = "2018"
+description = """
+Print numbers from FIRST to LAST, in steps of INCREMENT.
```
Comment (GrayJack) [inline seq/Cargo.toml]: Can you use the word "display" instead?

File: seq/src/cli.rs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+pub(crate) fn create_app<'a, 'b>() -> App<'a, 'b> {
+    App::new(crate_name!())
+        .version(crate_version!())
+        .author(crate_authors!())
+        .about(crate_description!())
+        .help_message("Display help information.")
+        .version_message("Display version information.")
+        .help_short("?")
+        .settings(&[ColoredHelp])
+        .arg(Arg::with_name("FIRST INCREMENT LAST").multiple(true))
```
Comment (GrayJack) [inline seq/src/cli.rs]: This arg is required, isn't it?

File: seq/src/cli.rs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        .arg(
+            Arg::with_name("SEPERATOR")
+                .short("s")
+                .long("separator")
+                .help("use STRING to separate numbers (default: \\n)")
```
Comment (GrayJack) [inline seq/src/cli.rs]: I think the default text would be better with square bracket to be similar to the normal default text that clap uses

File: seq/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+        process::exit(1);
+    }
+}
+
+fn find_operands(args: &[&str]) -> (f64, f64, f64) {
+    match args.len() {
+        1 => (1.0, 1.0, args[0].parse::<f64>().unwrap()),
+        2 => (args[0].parse().unwrap(), 1.0, args[1].parse().unwrap()),
+        _ => (args[0].parse().unwrap(), args[1].parse().unwrap(), args[2].parse().unwrap()),
+    }
```
Comment (GrayJack) [inline seq/src/main.rs]: This unwraps aren't ok here, a user may pass something that cannot be parsed into `f64`. The parse case where it is not a valid number should be handled.

File: seq/src/main.rs [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+}
+
+fn max_decimal_digits(args: &[&str]) -> usize {
+    args.iter().map(|v| v.len() - v.find(".").map(|pos| pos + 1).unwrap_or(v.len())).max().unwrap()
+}
+
+fn max_digits(args: &[&str]) -> usize {
+    args.iter().map(|v| v.find(".").unwrap_or(v.len())).max().unwrap()
+}
```
Comment (GrayJack) [inline seq/src/main.rs]: Are this unwrap never fail? If they never fail, can you add comments for why it don't fail? If they can  fail, it should pass the information forward or be handled on the site

**Review by GrayJack — COMMENTED** [review]
Thanks for your contribution!! :smile: 

Besides these comments, could you add seq to all workspace toml files so the CI can check all for all platforms possible?

File: seq/src/main.rs (hunk 5 above)
Comment (yehohanan7) [reply seq/src/main.rs]: That's right, it should never fail. I add a comment to make it explicit.

File: seq/src/main.rs (hunk 4 above)
Comment (yehohanan7) [reply seq/src/main.rs]: updated to exit with a useful error message

**Comment by yehohanan7** [comment]
> Thanks for your contribution!! 
> 
> Besides these comments, could you add seq to all workspace toml files so the CI can check all for all platforms possible?

@GrayJack all toml files updated, thanks!

**Comment by GrayJack** [comment]
Seems good

bors r+

**Comment by marcospb19** [comment]
@GrayJack
Does -f need a dedicated new issue?

**Comment by GrayJack** [comment]
Yes, probably is the best thing to do, I was just lazy these days

**Pull request merged**